pelican/plugins/sitemap/sitemap.py

```diff
@@ -55,235 +58,104 @@ def format_date(date):
-            fd.write(self.siteurl + "/" + pageurl + "\n")
-
-    def get_date_modified(self, page, default):
-        if hasattr(page, "modified"):
-            if isinstance(page.modified, datetime):
-                return page.modified
-            return get_date(page.modified)
-        else:
-            return default
-
-    def set_url_wrappers_modification_date(self, wrappers):
-        for (wrapper, articles) in wrappers:
-            lastmod = datetime.min.replace(tzinfo=self.timezone)
-            for article in articles:
-                lastmod = max(lastmod, article.date.replace(tzinfo=self.timezone))
-                try:
-                    modified = self.get_date_modified(article, datetime.min).replace(
-                        tzinfo=self.timezone
-                    )
-                    lastmod = max(lastmod, modified)
-                except ValueError:
-                    # Supressed: user will be notified.
-                    pass
-            setattr(wrapper, "modified", str(lastmod))
-
-    def generate_output(self, writer):
-        path = os.path.join(self.output_path, "sitemap.{0}".format(self.format))
-
-        pages = (
-            self.context["pages"]
-            + self.context["articles"]
-            + [c for (c, a) in self.context["categories"]]
-            + [t for (t, a) in self.context["tags"]]
-            + [a for (a, b) in self.context["authors"]]
-        )
-
-        self.set_url_wrappers_modification_date(self.context["categories"])
-        self.set_url_wrappers_modification_date(self.context["tags"])
-        self.set_url_wrappers_modification_date(self.context["authors"])
-
-        for article in self.context["articles"]:
-            pages += article.translations
-
-        info("writing {0}".format(path))
-
-        with open(path, "w", encoding="utf-8") as fd:
-
-            if self.format == "xml":
-                fd.write(XML_HEADER)
-            else:
-                fd.write(TXT_HEADER.format(self.siteurl))
-
-            FakePage = collections.namedtuple(
-                "FakePage", ["status", "date", "url", "save_as"]
+            changefreq = (
+                context.get("SITEMAP", {})
+                .get("changefreqs", {})
+                .get(content_type, CHANGEFREQ_DEFAULTS[content_type])
             )
-
-            for standard_page in self.context["DIRECT_TEMPLATES"]:
-                standard_page_url = self.context.get(
-                    "{}_URL".format(standard_page.upper())
-                )
-                standard_page_save_as = self.context.get(
-                    "{}_SAVE_AS".format(standard_page.upper())
-                )
-                fake = FakePage(
-                    status="published",
-                    date=self.now,
-                    url=standard_page_url or "{}.html".format(standard_page),
-                    save_as=standard_page_save_as or "{}.html".format(standard_page),
+            assert changefreq in VALID_CHANGEFREQS
```

Comment:
Same as before, `assert` for control flow is not a good practice.

Reply:
Goddamn "schema validation" is neither control flow nor (should be) any of my business. :laughing: